src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -1309,7 +1379,20 @@ private Collection<FunctionDescriptor> GetFunctionDescriptors(Collection<Functio
 #endif
                 };
 
-            return GetFunctionDescriptors(functions, descriptorProviders);
+            IEnumerable<FunctionMetadata> combinedFunctionMetadata = null;
+            if (proxies != null && proxies.Count != 0)
+            {
+                // Proxy routes will take precedence over http trigger functions and http trigger routes so they will be added first to the list of function descriptors.
```

Comment:
You're relying on the fact that elsewhere in InitializeHttpFunctions we're initializing routes in a first wins manner? I think it might be clearer to put the ordering logic where it is needed in InitializeHttpFunctions. Then you won't need the changes to this function.

E.g. you could order using `functions.OrderBy(p => p.Metadata.IsProxy)`